### ISSUE TITLE
A faster attention mask operation

### ISSUE THREAD

**Issue by boji123** [wenet-e2e/wenet#479]
https://github.com/wenet-e2e/wenet/blob/7f00996e018839f0e2bb8835ef2452896d74ef1e/wenet/transformer/attention.py#L84-L90

currently, attention requires twice mask fill operation, which is not effective enough

I suggest the method below:
```python
if mask is not None:
    mask = mask.unsqueeze(1)  # (batch, 1, *, time2)
    scores = scores + mask # add 0 as keep, add inf as mask
attn = torch.softmax(scores, dim=-1)  # (batch, head, time1, time2)
```

then the mask should be replaced as below before sending to attention module
```python
def get_attn_inf_mask(mask):
    '''
        chage mask for new attention mask mechanism.
            |1 0 0|    |0 -inf -inf|
            |1 1 0| => |0    0 -inf|
            |1 1 1|    |0    0    0|
    '''
    return torch.zeros(mask.shape, device=mask.device).masked_fill(~mask, -float('inf'))
```

the function get_attn_inf_mask can be called in encoder and decoder module
https://github.com/wenet-e2e/wenet/blob/7f00996e018839f0e2bb8835ef2452896d74ef1e/wenet/transformer/encoder.py#L154-L161
```python
        chunk_masks = add_optional_chunk_mask(embed_xs, masks,
                                              self.use_dynamic_chunk,
                                              self.use_dynamic_left_chunk,
                                              decoding_chunk_size,
                                              self.static_chunk_size,
                                              num_decoding_left_chunks)
        chunk_masks = get_attn_inf_mask(chunk_masks)
        for layer in self.encoders:
            xs, chunk_masks, _ = layer(embed_xs, chunk_masks, pos_emb, mask_pad)
```

same as decoder.

this update can improve the GPU utilization for about 5% (85%->90%) and speed up iteration for about 2%
should I create a PR ?

**Comment by robin1001** [comment]
Thanks. It may slightly faster as you mentioned, however, it's also requires a new function. I think it's better remain it unchanged.
